vernac/declare.ml

```diff
@@ -566,7 +567,22 @@ let is_unsafe_typing_flags flags =
   let open Declarations in
   not (flags.check_universes && flags.check_guarded && flags.check_positive)
 
-let declare_constant ?(local = Locality.ImportDefaultBehavior) ~name ~kind ~typing_flags ?user_warns cd =
+let loc_table = Summary.ref ~name:"constant-loc-table" Cmap_env.empty
+
+let get_loc kn = Cmap_env.find_opt kn !loc_table
+
+let loc_obj =
+  Libobject.declare_object @@
+  Libobject.superglobal_object "constant-loc-obj"
+    ~cache:(fun (kn,loc) ->
+        loc_table := Cmap_env.add kn loc !loc_table)
+    ~subst:(Some (fun (subst,(kn,loc)) -> Mod_subst.subst_constant subst kn, loc))
+    ~discharge:(fun x -> Some x)
```

Comment:
Note the discussion on https://github.com/coq/coq/pull/16261/ , in particular this approach requires fully loading a `.vo` to determine location for objects, so it seems to rule out many interesting use cases.

Note also that as of today, .glob files work well for jump to definition, coq-lsp 0.2.0 implements global jump to definition just fine using this approach.

A flaw both approaches have is that whitespace-only changes will alter the hash of the file, someone complained about this, but IMO fairly orthogonal.

[Also you can likely reuse the `loc` parts of #16261]

Reply:
Using glob files does sound like a good alternative.
Do you use them to find the references or just the definition points? 
Have you had issues with missing entries?
What do you do for definitions from the current file?

Reply:
cc @rtetley

Reply:
@SkySkimmer it works pretty well so far, didn't do a super rigorous test. For definitions in the current file, they are processed as they are parsed, anyways you want to do that for more meta-data (display doc, etc...)

I mean, you can have this new ad-hoc table that is kept using the `unit -> unit` libobject side effect, but it is neither useful nor very maintenable IMHO.

Reply:
We did use .glob files in Coqoon for the same reason, but that is a bit ugly because:
- glob files may not exist (eg -no-glob)
- glob files may not be installed
- the current file does not have a glob file (yet). Worse:
  - you can't write a glob file since in interactive mode you undo/re-execute, but files on the disk don't roll back
  -  so we were using the possibility to stream the glob info for the current file to the feedback channel and we were storing this info UI side, keep it updated on changes, etc

The approach here does not have the problems above, i.e. it works the same for other files or the current file and is kept in sync via the summary.

At the same time if you have executed a sentence, you also have already loaded all the .vo files that sentence needs, especially in continuous mode. So the advantage of not requiring to load .vo files seems marginal to me.